ClientDisconnectTest failures

https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-pr-builder/16979/testReport/com.hazelcast.client/ClientDisconnectTest/testPendingInvocationAndWaitEntryCancelled_whenDisconnected_withReliableTopic/

```
Error Message
Expected: an empty collection
     but: <[5=Invocation{op=com.hazelcast.ringbuffer.impl.operations.ReadManyOperation{serviceName='hz:impl:ringbufferService', identityHash=1817896872, partitionId=56, replicaIndex=0, callId=5, invocationTime=1535966623254 (2018-09-03 09:23:43.254), waitTimeout=-1, callTimeout=60000, name=_hz_rb_742b857f-405b-4a59-b00c-b2e7956a220c}, tryCount=250, tryPauseMillis=500, invokeCount=1, callTimeoutMillis=60000, firstInvocationTimeMs=1535966623254, firstInvocationTime='2018-09-03 09:23:43.254', lastHeartbeatMillis=0, lastHeartbeatTime='1970-01-01 00:00:00.000', target=[127.0.0.1]:5701, pendingResponse={VOID}, backupsAcksExpected=0, backupsAcksReceived=0, connection=null}]>
Stacktrace
java.lang.AssertionError: 

Expected: an empty collection
     but: <[5=Invocation{op=com.hazelcast.ringbuffer.impl.operations.ReadManyOperation{serviceName='hz:impl:ringbufferService', identityHash=1817896872, partitionId=56, replicaIndex=0, callId=5, invocationTime=1535966623254 (2018-09-03 09:23:43.254), waitTimeout=-1, callTimeout=60000, name=_hz_rb_742b857f-405b-4a59-b00c-b2e7956a220c}, tryCount=250, tryPauseMillis=500, invokeCount=1, callTimeoutMillis=60000, firstInvocationTimeMs=1535966623254, firstInvocationTime='2018-09-03 09:23:43.254', lastHeartbeatMillis=0, lastHeartbeatTime='1970-01-01 00:00:00.000', target=[127.0.0.1]:5701, pendingResponse={VOID}, backupsAcksExpected=0, backupsAcksReceived=0, connection=null}]>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at com.hazelcast.client.ClientDisconnectTest$8.run(ClientDisconnectTest.java:223)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1279)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1296)
	at com.hazelcast.client.ClientDisconnectTest.assertEmptyPendingInvocationAndWaitSet(ClientDisconnectTest.java:220)
	at com.hazelcast.client.ClientDisconnectTest.testPendingInvocationAndWaitEntryCancelled_whenDisconnected_withReliableTopic(ClientDisconnectTest.java:211)
```

--- a/hazelcast-client/src/test/java/com/hazelcast/client/ClientDisconnectTest.java
+++ b/hazelcast-client/src/test/java/com/hazelcast/client/ClientDisconnectTest.java
@@ -28,7 +28,6 @@ import com.hazelcast.core.IQueue;
 import com.hazelcast.core.Message;
 import com.hazelcast.spi.impl.NodeEngineImpl;
 import com.hazelcast.spi.impl.operationparker.impl.OperationParkerImpl;
-import com.hazelcast.spi.impl.operationservice.impl.Invocation;
 import com.hazelcast.spi.impl.operationservice.impl.InvocationRegistry;
 import com.hazelcast.spi.impl.operationservice.impl.OperationServiceImpl;
 import com.hazelcast.spi.properties.GroupProperty;
@@ -38,19 +37,17 @@ import com.hazelcast.test.HazelcastTestSupport;
 import com.hazelcast.test.annotation.ParallelTest;
 import com.hazelcast.test.annotation.QuickTest;
 import com.hazelcast.topic.ReliableMessageListener;
-import org.hamcrest.Matchers;
 import org.junit.After;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
 
-import java.util.Map;
 import java.util.concurrent.CountDownLatch;
 
 import static java.util.concurrent.TimeUnit.SECONDS;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
 
 @RunWith(HazelcastParallelClassRunner.class)
 @Category({QuickTest.class, ParallelTest.class})
@@ -190,7 +187,7 @@ public class ClientDisconnectTest extends HazelcastTestSupport {
             }
         });
 
-        sleepSeconds(3);
+        assertNonEmptyPendingInvocationAndWaitSet(server);
 
         client.shutdown();
 
@@ -206,9 +203,32 @@ public class ClientDisconnectTest extends HazelcastTestSupport {
         // ReliableTopic listener registers a blocking invocation
         client.getReliableTopic(randomName()).addMessageListener(new NopReliableMessageListener());
 
+        assertNonEmptyPendingInvocationAndWaitSet(server);
+
         client.shutdown();
 
         assertEmptyPendingInvocationAndWaitSet(server);
+    }
+
+    private void assertNonEmptyPendingInvocationAndWaitSet(HazelcastInstance server) {
+        NodeEngineImpl nodeEngine = getNodeEngineImpl(server);
+        OperationServiceImpl operationService = (OperationServiceImpl) nodeEngine.getOperationService();
+        final InvocationRegistry invocationRegistry = operationService.getInvocationRegistry();
+        final OperationParkerImpl operationParker = (OperationParkerImpl) nodeEngine.getOperationParker();
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() {
+                assertFalse(invocationRegistry.entrySet().isEmpty());
+            }
+        });
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() {
+                assertTrue(operationParker.getTotalParkedOperationCount() > 0);
+            }
+        });
     }
 
     private void assertEmptyPendingInvocationAndWaitSet(HazelcastInstance server) {
@@ -220,16 +240,16 @@ public class ClientDisconnectTest extends HazelcastTestSupport {
         assertTrueEventually(new AssertTask() {
             @Override
             public void run() {
-                assertThat(invocationRegistry.entrySet(), Matchers.<Map.Entry<Long, Invocation>>empty());
+                assertTrue(invocationRegistry.entrySet().isEmpty());
             }
-        }, 10);
+        });
 
         assertTrueEventually(new AssertTask() {
             @Override
             public void run() {
                 assertEquals(0, operationParker.getTotalParkedOperationCount());
             }
-        }, 10);
+        });
     }
 
     private static class NopReliableMessageListener implements ReliableMessageListener<Object> {

